dropwizard-metrics/dropwizard-metrics5/src/main/java/com/wavefront/integrations/metrics/WavefrontReporter.java

```diff
@@ -496,23 +525,35 @@ private void reportCounter(MetricName metricName, Counter counter) throws IOExce
     if (counter instanceof DeltaCounter) {
       long count = counter.getCount();
       String name = MetricConstants.DELTA_PREFIX + prefixAndSanitize(metricName.getKey().substring(1), "count");
-      wavefront.send(name, count,clock.getTime() / 1000, source, metricName.getTags());
+      wavefront.send(name, count,clock.getTime() / 1000, source, getMetricTags(metricName));
       counter.dec(count);
     } else {
-      wavefront.send(prefixAndSanitize(metricName.getKey(), "count"), counter.getCount(), clock.getTime() / 1000, source, metricName.getTags());
+      wavefront.send(prefixAndSanitize(metricName.getKey(), "count"), counter.getCount(), clock.getTime() / 1000, source, getMetricTags(metricName));
     }
   }
 
   private void reportGauge(MetricName metricName, Gauge<Number> gauge) throws IOException {
-    wavefront.send(prefixAndSanitize(metricName.getKey()), gauge.getValue().doubleValue(), clock.getTime() / 1000, source, metricName.getTags());
+    wavefront.send(prefixAndSanitize(metricName.getKey()), gauge.getValue().doubleValue(), clock.getTime() / 1000, source, getMetricTags(metricName));
   }
 
   private void sendIfEnabled(MetricAttribute type, MetricName metricName, double value, long timestamp) throws IOException {
     if (!getDisabledMetricAttributes().contains(type)) {
-      wavefront.send(prefixAndSanitize(metricName.getKey(), type.getCode()), value, timestamp, source, metricName.getTags());
+      wavefront.send(prefixAndSanitize(metricName.getKey(), type.getCode()), value, timestamp, source, getMetricTags(metricName));
     }
   }
 
+  private HashMap<String, String> getMetricTags(MetricName metricName) {
+    HashMap<String, String> metricTags = new HashMap<>(pointTags);
```

Comment:
Perhaps we can avoid creating a HashMap for cases when pointTags is empty and there are no metric level tags either:

```
tagCount = <pointTags size + metricName.getTags size> # with relevant null handling
if (tagCount == 0) {
  return Collections.emptyMap();
}
Map<String, String> metricTags = Maps.newHashMapWithExpectedSize(tagCount);
....
return metricTags;
```